### ISSUE TITLE
Moblin support for portrait mode streaming

### ISSUE THREAD

**Issue by fdebbabi** [eerimoq/moblin#88]
The stream via Moblin seems possible only in landscape mode. Any idea how to stream in portrait mode  
 from within the app?

My hack for now is to send it to a middle component that is going to do a rotate for something that should be native I believe.

This is extremely beneficial for portrait live streams that target smartphones so that viewers can see a bigger and sharper image and not simply a 16:9 image in a 9:16 canvas (a big part of the frame is black) (check added photo to see how the stream looks on smartphones screens without doing the hack

<img width="384" alt="Screenshot 2024-12-15 at 14 19 35" src="https://github.com/user-attachments/assets/5ce367ac-b7f0-40d6-9d62-9b7c61ee2ef0" />
)

**Comment by eerimoq** [comment]
Settings -> Streams -> My stream -> Portrait not sufficient, and rotate 90 degrees in OBS?

**Comment by fdebbabi** [comment]
As I stream outdoor for 1-5 hours, my goal is to skip OBS part (that I am doing currently), my goal is to send stream following the **first** approach as there is less dependency/requirements/things to manage at home and when things fail when I am outside I don't have to stop livestreaming and go back at home and fix things:

1. Moblin(SRTLA) -> Belabox(SRTLA) **<-** RTMP Relay -> (Youtube, Tiktok and Instagram simultaneously)

2. Moblin(SRTLA) -> Belabox(SRTLA) **<-** OBS On my PC to rotate -> RTMP Relay -> (Youtube, Tiktok and Instagram simultaneously)

**Comment by eerimoq** [comment]
I see. Real portrait is not someone is working on, so likely not supported anytime soon

**Comment by fdebbabi** [comment]
Ok, thank you for the information 👍

**Comment by eerimoq** [comment]
Implemented it. It's in Test Flight soon.

**Comment by fdebbabi** [comment]
That was extremly fast 😅 you know that portrait streaming is the only reason I keep using Larix Broadcaster (I bet a lof people need this for IRL)

**Comment by eerimoq** [comment]
It's in test flight now in case you wanna give it a try. Just enable it in Settings -> Streams -> My stream and go live. Let me know if it works

**Comment by fdebbabi** [comment]
I have just checked (on iPad), one question, how to enable it in Settings -> Streams -> My stream, I can't find a button to rotate stream or proper 9:16 resolution, i only find rotate in display.

**Comment by eerimoq** [comment]
it's only supported on iphone

**Comment by eerimoq** [comment]
can probably be added to ipad

**Comment by eerimoq** [comment]
was easy to make it work on ipad. will be in test flight in 15 minutes

**Comment by fdebbabi** [comment]
Stream is working fine, some minor issues I sent via testFlight

**Comment by eerimoq** [comment]
I need more details on the problem that parts of the video is outside the screen. It's not like that for me on iPhone or iPad. What is you setup? What camera are you using as video source?

**Comment by eerimoq** [comment]
fixed the resolution names

**Comment by eerimoq** [comment]
looks good on iphone x as well. not sure why the video is partly outside the screen. you got any widgets or anything else you think could affect it?

**Comment by eerimoq** [comment]
Update to latest testflight again. should have resolution fix

**Comment by fdebbabi** [comment]
Yes, all of the issues mentioned are now fixed on both ipad and iPhone SE (even the camera that was out of the screen in portrait mode).
One minor issue still left is that icons in portrait mode are slightly cut/out of the screen in iPHone SE (I sent photo using test flight)

**Comment by eerimoq** [comment]
ok! yeah, the buttons are slightly cut on Mac as well. not sure how to fix it, as I don't want to add padding on phones where it already looks good.

**Comment by fdebbabi** [comment]
Yes, for me everything is perfect :) that would be good also to share it in discord once accessible directly in app store (multiple people were asking also for it)

**Comment by eerimoq** [comment]
fixed the buttons as well. was easy.

**Comment by eerimoq** [comment]
You can read the release notes in moblin discord server, the releases channel.

**Comment by eerimoq** [comment]
it's in test flight now

**Comment by fdebbabi** [comment]
I will check that tomorrow

**Comment by fdebbabi** [comment]
Buttons are working fine now, portrait mode seems not working when using external camera (DJI Action 4) in iPad, i didn't check the resolution received by the stream, for now I am speaking only about the UI, i believe it should rotate. 
![20241219_110258](https://github.com/user-attachments/assets/41b402fa-d39f-4ba1-aa16-f538125084d6)

**Comment by eerimoq** [comment]
Added rotation setting in Settings -> Scenes -> My scene just now. It's in test flight.

**Comment by fdebbabi** [comment]
Now it rotates properly, but when in portrait mode it does not fill enough space in the screen, it would perfect if it is a little bit bigger 
![20241220_142724](https://github.com/user-attachments/assets/24bd6eaa-b49c-46f1-a1ad-825604432413)

**Comment by eerimoq** [comment]
odd it's not like that for me

**Comment by fdebbabi** [comment]
I on iPaid Air Model, I will in chech again one hour

**Comment by fdebbabi** [comment]
Everything works fine now 🤩

**Comment by eerimoq** [comment]
Goood!

**Comment by fdebbabi** [comment]
One question, when using external camera, does moblin decodes the received stream from the camera to show it on screen, than re-encodes the decoded one to send it to the destination, or does it send as it is from the source?

**Comment by eerimoq** [comment]
Everything leaving Moblin is encoded by Moblin.
